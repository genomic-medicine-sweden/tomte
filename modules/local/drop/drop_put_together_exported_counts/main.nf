process DROP_PUT_TOGETHER_EXPORTED_COUNTS {
    tag "DROP_put_together_exported_counts"
    label 'process_low'

    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        exit 1, "Local DROP module does not support Conda. Please use Docker / Singularity / Podman instead."
    }

    container "docker.io/clinicalgenomics/drop:1.4.0"

    input:
    path(exported_counts_ae)
    path(exported_counts_as)
    tuple val(meta), path(gtf)

    output:
    path('exported_counts'), emit: drop_exported_counts
    path "versions.yml"    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def ae_run = exported_counts_ae ? true : false
    def as_run = exported_counts_as ? true : false
    def gtf_no_extension = gtf.baseName

    """

    mkdir -p exported_counts
    if [[ "$ae_run" == "true" ]];then
        cp ${exported_counts_ae}/* exported_counts/.
    fi

    if [[ "$as_run" == "true" ]];then
        cp ${exported_counts_as}/* exported_counts/.
    fi

    mv exported_counts/sample_annotation.tsv .

    drop_sample_annot_exported_counts.py \\
        --sample_annot "sample_annotation.tsv" \\
        --ae_run $ae_run \\
        --as_run $as_run \\
        --gtf $gtf_no_extension

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        drop_sample_annotation_exported_counts: \$(drop_sample_annotation_exported_counts.py --version)
    END_VERSIONS

    """

    stub:
    """
    mkdir -p exported_counts

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        drop_sample_annotation_exported_counts: \$(drop_sample_annotation_exported_counts.py --version)
    END_VERSIONS
    """
}
