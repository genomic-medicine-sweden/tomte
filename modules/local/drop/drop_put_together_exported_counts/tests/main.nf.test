nextflow_process {

    name "Test Process DROP_PUT_TOGETHER_EXPORTED_COUNTS"
    script "../main.nf"
    process "DROP_PUT_TOGETHER_EXPORTED_COUNTS"
    config "./nextflow.config"
    tag "modules"
    tag "modules_local"
    tag "drop"
    tag "drop_put_together_exported_counts"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
            script "../../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'fasta' ],
                    file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("SAMTOOLS_FAIDX") {
            script "../../../../../modules/nf-core/samtools/faidx/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = [[],[]]
                input[2] = []
                """
            }
        }

        run("GUNZIP", alias: "GUNZIP_GTF") {
            script "../../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'gtf' ],
                    file("${projectDir}/test_data/grch37_chr21.gtf.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("DROP_SAMPLE_ANNOT") {
            script "../../../../../modules/local/drop/drop_sample_annot/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = Channel.of([
                            ['ACC5963A1'], [false], ['reverse'], ['NA'], ["${projectDir}/test_data/test_ACC5963A1.vcf.gz"], ["${projectDir}/test_data/test_ACC5963A1.vcf.gz.tbi"],
                            file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                            file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                        ])
                input[2] = channel.fromPath("${projectDir}/test_data/drop_data/geneCounts.tsv.gz", checkIfExists: true)
                input[3] = channel.fromPath("${projectDir}/test_data/drop_data/sampleAnnotation.tsv", checkIfExists: true)
                input[4] = 'outrider'
                input[5] = 'fraser'
                """
            }
        }

        run("DROP_CONFIG_RUN_AS") {
            script "../../../../../modules/local/drop/drop_config_runAS/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip.join(SAMTOOLS_FAIDX.out.fai).collect()
                input[1] = GUNZIP_GTF.out.gunzip.collect()
                input[2] = DROP_SAMPLE_ANNOT.out.drop_annot
                input[3] = Channel.of([
                    file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                    ])
                input[4] = channel.fromPath("${projectDir}/test_data/drop_data/geneCounts.tsv.gz", checkIfExists: true)
                input[5] = channel.fromPath("${projectDir}/test_data/drop_data", checkIfExists: true)
                input[6] = 'GRCh37'
                input[7] = 'fraser'
                input[8] = 'outrider'
                input[9] = 1
                input[10] = false
                """
            }
        }

        run("DROP_CONFIG_RUN_AE") {
            script "../../../../../modules/local/drop/drop_config_runAE/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip.join(SAMTOOLS_FAIDX.out.fai).collect()
                input[1] = GUNZIP_GTF.out.gunzip.collect()
                input[2] = DROP_SAMPLE_ANNOT.out.drop_annot
                input[3] = Channel.of([
                        file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                    ])
                input[4] = channel.fromPath("${projectDir}/test_data/drop_data/geneCounts.tsv.gz", checkIfExists: true)
                input[5] = channel.fromPath("${projectDir}/test_data/drop_data", checkIfExists: true)
                input[6] = 'GRCh37'
                input[7] = 'outrider'
                input[8] = 'fraser'
                input[9] = 1
                input[10] = 2.5
                input[11] = false
                """
            }
        }

    }

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = DROP_CONFIG_RUN_AE.out.gene_counts_ae
                input[1] = DROP_CONFIG_RUN_AS.out.gene_counts_as
                input[2] = GUNZIP_GTF.out.gunzip.collect()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.drop_exported_counts.collect{ file(it).name },
                    process.out.versions,
                ).match()}
            )
        }

    }

}
