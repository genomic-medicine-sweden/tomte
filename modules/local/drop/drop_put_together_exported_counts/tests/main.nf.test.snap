{
    "Should run without failures": {
        "content": [
            [
                "exported_counts"
            ],
            [
                "versions.yml:md5,ef2c34df7d7aa59f1effd1f590cd4582"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-08-12T21:47:03.176647877"
    }
}